core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -291,11 +291,20 @@ private[spark] object TaskMetrics extends Logging {
 
 private[spark] class BlockStatusesAccumulator
   extends AccumulatorV2[(BlockId, BlockStatus), Seq[(BlockId, BlockStatus)]] {
-  private[this] var _seq = ArrayBuffer.empty[(BlockId, BlockStatus)]
+  private var _seq = ArrayBuffer.empty[(BlockId, BlockStatus)]
 
   override def isZero(): Boolean = _seq.isEmpty
 
-  override def copyAndReset(): BlockStatusesAccumulator = new BlockStatusesAccumulator
+  override def copy(): BlockStatusesAccumulator = {
```

Comment:
the `copyAndReset` version is much cheaper than calling `copy` and then `reset`, how about we just add a new `reset` method?

cc @rxin